src/charm.py

```diff
@@ -976,70 +1056,106 @@ def _core_reconciliation(self):
-            raise RuntimeError(
-                f"wp theme list command failed, stdout is not json: {repr(process.stdout)}"
+            return self._ExecResult(
+                success=False,
+                result=None,
+                message=f"wp {addon_type} list command failed, stdout is not json"
             )
 
-    def _wp_theme_install(self, theme):
-        """Install WordPress theme"""
-        process = self._run_wp_cli(
+    def _wp_addon_install(self, addon_type, addon_name):
+        """Install WordPress addon (plugin/theme)"""
+        self._check_addon_type(addon_type)
+        if addon_type == "theme":
             # --force will overwrite any installed version of the theme,
             # without prompting for confirmation
-            ["wp", "theme", "install", theme, "--force"],
-            timeout=600,
-            combine_stderr=True
-        )
-        if process.return_code != 0:
-            raise RuntimeError("wp theme list command failed: " + process.stdout)
-
-    def _wp_theme_delete(self, theme):
-        """Uninstall WordPress theme"""
-        process = self._run_wp_cli(
-            ["wp", "theme", "delete", theme, "--force"],
-            combine_stderr=True
-        )
-        if process.return_code != 0:
-            raise RuntimeError("wp theme list command failed: " + process.stdout)
+            cmd = ["wp", "theme", "install", addon_name, "--force"]
+        else:
+            cmd = ["wp", "plugin", "install", addon_name]
+        return self._wrapped_run_wp_cli(cmd, timeout=600)
+
+    def _wp_addon_uninstall(self, addon_type, addon_name):
+        """Uninstall WordPress addon (theme/plugin)"""
+        self._check_addon_type(addon_type)
+        if addon_type == "theme":
+            cmd = ["wp", "theme", "delete", addon_name, "--force"]
+        else:
+            cmd = ["wp", "plugin", "uninstall", addon_name, "--deactivate"]
+        return self._wrapped_run_wp_cli(cmd, timeout=600)
 
-    def _theme_reconciliation(self):
-        """Reconciliation process for WordPress themes
+    def _addon_reconciliation(self, addon_type):
+        """Reconciliation process for WordPress addons (theme/plugin)
 
-        Install and uninstall themes to match the themes setting in config
+        Install and uninstall themes/plugins to match the themes/plugins setting in config
         """
-        logger.debug("Start theme reconciliation process")
-        current_installed_themes = set(t["name"] for t in self._wp_theme_list())
-        logger.debug("Currently installed themes %s", current_installed_themes)
-        themes_in_config = [
-            t.strip() for t in self.model.config["themes"].split(",")
+        self._check_addon_type(addon_type)
+        logger.debug(f"Start {addon_type} reconciliation process")
+        current_installed_addons = set(t["name"] for t in self._wp_addon_list(addon_type).result)
+        logger.debug(f"Currently installed {addon_type}s %s", current_installed_addons)
```

Comment:
https://github.com/canonical/is-charms-contributing-guide/blob/main/CONTRIBUTING.md#formatting-log-messages

Reply:
Same as above.

Reply:
This was also referring to the use of f-strings in logging messages which is a security issue